### PR TITLE
Add Groq LLM integration

### PR DESCRIPTION
**Pull request by gradenr**
# Description
Adds a LLM integration for Groq models.

Fixes # (issue)

## Type of Change

Please delete options that are not relevant.

- [X] New feature (non-breaking change which adds functionality)
- [X] This change requires a documentation update

# How Has This Been Tested?

- [X] Added new unit/integration tests
- [X] Added new notebook (that tests end-to-end)
- [X] I stared at the code and made sure it makes sense

# Suggested Checklist:

- [X] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation
- [?] I have added Google Colab support for the newly added notebooks.
- [X] My changes generate no new warnings
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes
- [X] I ran `make format; make lint` to appease the lint gods

### CODE REVIEW

**Review by hatianzhang — APPROVED** [review]
nice!

**Review by anoopshrma — APPROVED** [review]

File: llama-index-integrations/llms/llama-index-llms-groq/pyproject.toml [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+check-filenames = true
+check-hidden = true
+skip = "*.csv,*.html,*.json,*.jsonl,*.pdf,*.txt,*.ipynb"
+
+[tool.llamahub]
+contains_example = false
+import_path = "llama_index.llms.groq"
+
+[tool.llamahub.class_authors]
+groq = "llama-index"
```
Comment (nerdai) [inline llama-index-integrations/llms/llama-index-llms-groq/pyproject.toml:15]: feel free to add your Github user name here :)

Comment (nerdai) [reply llama-index-integrations/llms/llama-index-llms-groq/pyproject.toml:15]: Also this should match the class name, which I believe is `Groq`

**Review by nerdai — APPROVED** [review]
lgtm! I left some minor comments re: llamahub metadata

File: llama-index-integrations/llms/llama-index-llms-groq/pyproject.toml (hunk 1 above)
Comment (gradenr) [reply llama-index-integrations/llms/llama-index-llms-groq/pyproject.toml]: Updated!  Thanks for the feedback.

**Comment by nerdai** [comment]
3x approvals now :) Going to merge .

**Comment by nerdai** [comment]
published to [pypi](https://pypi.org/project/llama-index-llms-groq/) :)

**Pull request merged**